pivottable/index.js

```diff
@@ -0,0 +1,57 @@
+window.onerror = (err) => {
+  console.trace();
+  alert(String(err));
+};
+
+grist.ready({
+  requiredAccess: 'read table'
+});
+
+const urlParams = new URLSearchParams(window.location.search);
+const rows = urlParams.get('rows')?.split(',');
+const cols = urlParams.get('cols')?.split(',');
+const vals = urlParams.get('vals')?.split(',');
+const aggregatorName = urlParams.get('aggregatorName');
+const rendererName = urlParams.get('rendererName');
+
+function wavg (n) {
+  if (!n) { return; }
+  n = n.filter(([note]) => typeof (note) === 'number');
+  if (n.length) { return n.map(([note, coef]) => note * coef).reduce((a, b) => a + b) / n.map(([_note, coef]) => coef).reduce((a, b) => a + b); }
+}
+
+function weightedAverage ([val, coef]) {
+  return (_data, _rowKey, _colKey) => ({
+    values: [],
+    push: function (rec) { this.values.push([rec[val], rec[coef]]); },
+    value: function () { return wavg(this.values); },
+    format: function (x) { return (Math.round(x * 100) / 100).toFixed(2); },
+    numInputs: 2
+  });
+}
+
+const aggregators = {
+  'Weighted Average': weightedAverage
+};
+
+grist.onRecords(rec => {
+  console.log(rec);
+  $('#table').pivotUI(
+    rec,
+    {
+      rows,
+      cols,
+      vals,
+      aggregatorName,
+      rendererName,
+      aggregators: $.extend(aggregators, $.pivotUtilities.aggregators),
```

Comment:
What do you think about reversing the order, as built-in aggregations seem to be more commonly used by others?

```js
aggregators: $.extend($.pivotUtilities.aggregators, aggregators),
```